frontend/src/lib/k8s/apiProxy.test.ts

```diff
@@ -0,0 +1,1113 @@
+        });
+      });
+
+      await Promise.all(connections);
+      expect(cb).toHaveBeenCalledTimes(3 * mockMultipleResource.length);
+    });
+  });
+
+  describe('simpleApiFactoryWithNamespace', () => {
+    it('Successfully creates API client for single namespaced resource', done => {
+      mockServer = new WS(
+        `${wsUrl}/clusters/${clusterName}/apis/${mockSingleResource[0]}/${mockSingleResource[1]}/namespaces/${namespace}/${mockSingleResource[2]}`
+      );
+
+      const client = apiProxy.apiFactoryWithNamespace(...mockSingleResource);
+      expect(client.isNamespaced).toBe(true);
+
+      client.list(namespace, cb, errCb, {}, clusterName);
+
+      mockServer.on('connection', async (socket: any) => {
+        expect(cb).toHaveBeenNthCalledWith(1, []);
+
+        socket.send(JSON.stringify({ type: 'ADDED', object: mockConfigMap }));
+        await new Promise(resolve => setTimeout(resolve, 100));
+        expect(cb).toHaveBeenNthCalledWith(2, [{ actionType: 'ADDED', ...mockConfigMap }]);
+
+        socket.send(JSON.stringify({ type: 'MODIFIED', object: modifiedConfigMap }));
+        await new Promise(resolve => setTimeout(resolve, 100));
+        expect(cb).toHaveBeenNthCalledWith(3, [{ actionType: 'MODIFIED', ...modifiedConfigMap }]);
+
+        done();
+      });
+    });
+  });
+
+  describe('multipleApiFactoryWithNamespace', () => {
+    it('Successfully creates API client for multiple namespaced resources', async () => {
+      const connections = mockMultipleResource.map(([group, version, resource]) => {
+        return new Promise<void>(resolve => {
+          mockServers[resource] = new WS(
+            `${wsUrl}/clusters/${clusterName}/apis/${group}/${version}/namespaces/${namespace}/${resource}`
+          );
+          const client = apiProxy.apiFactoryWithNamespace(group, version, resource);
+          client.list(namespace, cb, errCb, {}, clusterName);
+
+          mockServers[resource].on('connection', async (socket: any) => {
+            expect(cb).toHaveBeenCalledWith([]);
+
+            socket.send(JSON.stringify({ type: 'ADDED', object: mockConfigMap }));
+            await new Promise(resolve => setTimeout(resolve, 100));
+            expect(cb).toHaveBeenCalledWith([{ actionType: 'ADDED', ...mockConfigMap }]);
+
+            socket.send(JSON.stringify({ type: 'MODIFIED', object: modifiedConfigMap }));
+            await new Promise(resolve => setTimeout(resolve, 100));
+            expect(cb).toHaveBeenCalledWith([{ actionType: 'MODIFIED', ...modifiedConfigMap }]);
+
+            resolve();
+          });
+        });
+      });
+
+      await Promise.all(connections);
+      expect(cb).toHaveBeenCalledTimes(3 * mockMultipleResource.length);
+    });
+  });
+});
+
+describe('request, post, patch, put, delete', () => {
+  const testData = { key: 'value' };
+
+  beforeEach(() => {
+    nock(baseApiUrl)
+      // Successful GET request
+      .persist()
+      .get(`/clusters/${clusterName}` + testPath)
```

Comment:
Can you please use template literals for this? (and in other places)

Like this:
```
`/clusters/${clusterName}${testPath}`
```
Rather than:
```
`/clusters/${clusterName}` + testPath
```

Reply:
adding in the next push